src/stochastic.jl

```diff
@@ -74,13 +123,18 @@ end
 
 # A time period with scenario number and probability
 struct ScenarioPeriod{P} <: TimePeriod where {P<:TimePeriod}
-    sc::Int
+    sc::Integer
     prob::Float64
     multiple::Float64
     period::P
 end
 
-function ScenarioPeriod(scenario::Int, prob::Number, multiple::Number, period)
+function ScenarioPeriod(
+    scenario::Integer,
+    prob::Number,
```

Comment:
Maybe we should use `Real` here instead of `Number`? Allowing e.g. `Complex` here makes no sense.

Reply:
I would guess that real makes a lot of sense for all cases where we have `Number` at the time being. None of the cases should support `Complex`.